### PR TITLE
Adds test project and ruby/cucumber type json output

### PR DESCRIPTION
**Pull request by sidekickr**
This is a large contribution I had talked with Ahmed about a few months ago. it got put on our back burner for some other work, but here it is.

We've created an official test project for Cucumberish, moving the features definition target and the CucumberishTest target from the examples project to this project. This cleans up examples to focus on being examples and gives a project for running unit tests against the library to check for regressions.

There is a failing test in the CucumberTest, but I confirmed that it was failing on the develop branch before generating the pull request.

Also in this submission (it would have been difficult to pull it out for a separate pull request) is JSON output for the test results in the format of ruby/cucumber. There are a few differences, in that we are using the cucumber JSON expanded formatter (https://gist.github.com/blt04/9866357) and are leaving out some of the output that doesn't make sense in objective-c.

I didn't do a full update on the README file. I'll work on that over the weekend and submit that separately.

### CODE REVIEW

File: Cucumberish/Core/Models/CCIJSONDumper.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+#import <Foundation/Foundation.h>
+@class CCIFeature;
+@interface CCIJSONDumper : NSObject
+
+/*!
+ *   @author Erik Call Reynolds 
+ *@date, 16-11-15 12:11:00
+ *
+ *   @brief creates the Cucumber JSON Output String similar to
```
Comment (brentleyjones) [inline Cucumberish/Core/Models/CCIJSONDumper.h]: Incomplete sentence

File: Cucumberish/Core/Models/CCIJSONDumper.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+//  Copyright © 2016 Ahmed Ali. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+@class CCIFeature;
+@interface CCIJSONDumper : NSObject
+
+/*!
+ *   @author Erik Call Reynolds 
+ *@date, 16-11-15 12:11:00
```
Comment (brentleyjones) [inline Cucumberish/Core/Models/CCIJSONDumper.h]: Indentation

File: Cucumberish/Cucumberish.m [hunk 3]
```diff
@@ -304,20 +305,30 @@ + (void)swizzleTestCaseWithSelectorImplementationForClass:(Class)class
...
 }
+
 + (NSInvocation *)invocationForScenarioOutline:(CCIScenarioDefinition *)outline exampleIndex:(NSInteger)index feature:(CCIFeature *)feature featureClass:(Class)featureClass
 {
     //Scenario for each body
     CCIScenarioDefinition * scenario = [outline copy];
-    scenario.name = [self exampleScenarioNameForScenarioName:scenario.name exampleAtIndex:index];
-    scenario.keyword = (NSString *)kScenarioOutlineKeyword;
+
+		scenario.keyword = (NSString *)kScenarioOutlineKeyword;
```
Comment (brentleyjones) [inline Cucumberish/Cucumberish.m]: Tabs when it should be spaces.

File: Cucumberish/Cucumberish.m [hunk 4]
```diff
@@ -304,20 +305,30 @@ + (void)swizzleTestCaseWithSelectorImplementationForClass:(Class)class
...
     scenario.examples = nil;
     CCIExample * example = outline.examples.firstObject;
+		scenario.name = [self exampleScenarioNameForScenarioName:scenario.name exampleAtIndex:index example:example];
```
Comment (brentleyjones) [inline Cucumberish/Cucumberish.m]: Tabs when it should be spaces.

File: Cucumberish/Cucumberish.m [hunk 5]
```diff
@@ -337,7 +348,9 @@ + (NSInvocation *)invocationForScenarioOutline:(CCIScenarioDefinition *)outline
             }
         }
     }
-    
+
+		[outline addOutlineChildScenario:scenario];
```
Comment (brentleyjones) [inline Cucumberish/Cucumberish.m]: Tabs when it should be spaces.

**Comment by Ahmed-Ali** [comment]
Guys... you are doing an amazing work over here!
It makes me feel guilty not to check it immediately :)
I mentioned it before (and mentioning it here in case you don't know about it @sidekickr), that I am having some personal obligations that might keep me so busy till the end of January.
Once I am done with these things, I will carefully check all the current issues and PRs and act on them immediately.
I am really sorry for not being able to do that immediately, and I can't thank you enough guys!
Please keep up this awesome work, and let's bring this library to its utmost value.

**Comment by sidekickr** [comment]
updated the file formatting and documentation per comments by @brentleyjones

**Comment by brentleyjones** [comment]
This is really neat btw!

**Comment by Ahmed-Ali** [comment]
Has been merged to develop and going to master soon. Thanks guys for the amazing work!

**Pull request merged**